### PR TITLE
alacritty: use $TERMINFO if set in environment to install terminfo files

### PR DESCRIPTION
**Pull request by imomaliev**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/alacritty.rb [hunk 1]
```diff
@@ -21,9 +21,9 @@
   binary "#{appdir}/Alacritty.app/Contents/Resources/completions/alacritty.fish",
          target: "#{HOMEBREW_PREFIX}/share/fish/vendor_completions.d/alacritty.fish"
   binary "#{appdir}/Alacritty.app/Contents/Resources/61/alacritty",
-         target: "#{ENV["HOME"]}/.terminfo/61/alacritty"
+         target: "#{ENV["TERMINFO"] || "#{ENV["HOME"]}/.terminfo"}/61/alacritty"
```
Comment (carlocab) [inline Casks/alacritty.rb:24]:
```suggestion
         target: "#{ENV.fetch("TERMINFO", "#{ENV["HOME"]}/.terminfo")}/61/alacritty"
```

Comment (carlocab) [reply Casks/alacritty.rb:24]: Actually, does this work for you? `TERMINFO` is likely sanitised from the environment before this code is run.

Comment (imomaliev) [reply Casks/alacritty.rb]: @carlocab How can I ran this Cask localy?

File: Casks/alacritty.rb [hunk 2]
```diff
@@ -21,9 +21,9 @@
   binary "#{appdir}/Alacritty.app/Contents/Resources/completions/alacritty.fish",
          target: "#{HOMEBREW_PREFIX}/share/fish/vendor_completions.d/alacritty.fish"
   binary "#{appdir}/Alacritty.app/Contents/Resources/61/alacritty",
-         target: "#{ENV["HOME"]}/.terminfo/61/alacritty"
+         target: "#{ENV.fetch("TERMINFO", "#{ENV["HOME"]}/.terminfo"}/61/alacritty"
```
Comment (carlocab) [inline Casks/alacritty.rb:24]:
```suggestion
         target: "#{ENV.fetch("TERMINFO", "#{ENV["HOME"]}/.terminfo")}/61/alacritty"
```

File: Casks/alacritty.rb [hunk 3]
```diff
@@ -21,9 +21,9 @@
...
   binary "#{appdir}/Alacritty.app/Contents/Resources/61/alacritty-direct",
-         target: "#{ENV["HOME"]}/.terminfo/61/alacritty-direct"
+         target: "#{ENV.fetch("TERMINFO", "#{ENV["HOME"]}/.terminfo"}/61/alacritty-direct"
```
Comment (carlocab) [inline Casks/alacritty.rb:26]:
```suggestion
         target: "#{ENV.fetch("TERMINFO", "#{ENV["HOME"]}/.terminfo")}/61/alacritty-direct"
```

File: Casks/alacritty.rb (hunk 1 above)
Comment (carlocab) [reply Casks/alacritty.rb]: You can do something like
```
cd "$(brew --repo homebrew/cask)"
gh pr checkout 119610
brew reinstall alacritty
```

**Comment by imomaliev** [comment]
Installation currently fails

**Comment by imomaliev** [comment]
```
brew install --cask alacritty
Error: Cask 'alacritty' is unreadable: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/alacritty.rb:24: syntax error, unexpected tSTRING_DEND, expecting ')'
...O", "#{ENV["HOME"]}/.terminfo"}/61/alacritty"
...                              ^
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/alacritty.rb:24: syntax error, unexpected tSTRING_BEG, expecting do or '{' or '('
...ME"]}/.terminfo"}/61/alacritty"
...                              ^
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/alacritty.rb:26: syntax error, unexpected ':', expecting tSTRING_DEND
         target: "#{ENV.fetch("TERMINFO", "#{...
               ^
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/alacritty.rb:26: syntax error, unexpected tSTRING_DEND, expecting ')'
...O", "#{ENV["HOME"]}/.terminfo"}/61/alacritty-direct"
...                              ^
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/alacritty.rb:26: syntax error, unexpected tSTRING_BEG, expecting do or '{' or '('
...terminfo"}/61/alacritty-direct"
...                              ^
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/alacritty.rb:34: syntax error, unexpected end, expecting tSTRING_DEND
```

I have but if I use `ENV["TERMINFO"] || "#{ENV["HOME"]}/.terminfo"` it works correctly

**Comment by imomaliev** [comment]
I also when I use `Utils` https://docs.brew.sh/Cask-Cookbook#arbitrary-ruby-methods `brew styles` gives me 

```
brew style Casks/alacritty.rb
Casks/alacritty.rb:2:3: W: Do not define constants this way within a block.
  module Utils ...
```

**Comment by carlocab** [comment]
> Installation currently fails
> 
> 
> 
> ```
> 
> brew install --cask alacritty
> 
> Error: Cask 'alacritty' is unreadable: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/alacritty.rb:24: syntax error, unexpected tSTRING_DEND, expecting ')'
> 
> ...O", "#{ENV["HOME"]}/.terminfo"}/61/alacritty"
> 
> ...                              ^
> 
> /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/alacritty.rb:24: syntax error, unexpected tSTRING_BEG, expecting do or '{' or '('
> 
> ...ME"]}/.terminfo"}/61/alacritty"
> 
> ...                              ^
> 
> /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/alacritty.rb:26: syntax error, unexpected ':', expecting tSTRING_DEND
> 
>          target: "#{ENV.fetch("TERMINFO", "#{...
> 
>                ^
> 
> /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/alacritty.rb:26: syntax error, unexpected tSTRING_DEND, expecting ')'
> 
> ...O", "#{ENV["HOME"]}/.terminfo"}/61/alacritty-direct"
> 
> ...                              ^
> 
> /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/alacritty.rb:26: syntax error, unexpected tSTRING_BEG, expecting do or '{' or '('
> 
> ...terminfo"}/61/alacritty-direct"
> 
> ...                              ^
> 
> /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/alacritty.rb:34: syntax error, unexpected end, expecting tSTRING_DEND
> 
> ```

You were missing a closing `)` here.

**Comment by imomaliev** [comment]
Oh, I was using previous commit in cask folder. It is ready for review and working

**Comment by imomaliev** [comment]
@carlocab Do you know why `brew style` could complain about `module Utils`? Also this ready to be merged.

**Comment by miccal** [comment]
Thank you @imomaliev.

**Comment by imomaliev** [comment]
@miccal Thank you. And big thanks to @carlocab for reviewing and helping with this PR

**Comment by carlocab** [comment]
Thanks for working on this, @imomaliev. Not sure about your question re `brew style`, though.

**Pull request merged**